picosha2.h

```diff
@@ -369,7 +373,43 @@ template <typename InContainer>
 std::string hash256_hex_string(const InContainer& src) {
     return hash256_hex_string(src.begin(), src.end());
 }
-
-}  // namespace picosha2
-
+std::string hash256_from_ifstream(std::ifstream f){
+    std::vector<char> data;
+    data.insert(data.begin(),std::istreambuf_iterator<char>(f), std::istreambuf_iterator<char>());
+    std::string hashed;
+    hash256_hex_string(std::string(data.begin(),data.end()), hashed);
+    return hashed;
+}
+inline void multithread(std::vector<std::string> src, std::deque<std::vector<std::string>>* solved){
+    std::vector<std::string> solve;
+    for(std::string i: src){
+        solve.push_back(hash256_hex_string(i));
+    }
+    solved->push_back(solve);
+}
+std::vector<std::string> hash_Vector(std::vector<std::string> src){
+    int threads_num = std::thread::hardware_concurrency();
```

Comment:
FYI `std::thread::hardware_concurrency()` can return 0. It occurs 0 division error later. see https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency